packages/p2p/src/components/app-content.jsx

```diff
@@ -13,9 +14,32 @@ import NicknameForm from './nickname-form';
 import Orders from './orders/orders.jsx';
 import TemporarilyBarredHint from './temporarily-barred-hint';
 import Verification from './verification/verification.jsx';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
 
-const AppContent = () => {
+const AppContent = ({ order_id }) => {
     const { buy_sell_store, general_store } = useStores();
+    const { showModal, hideModal } = useModalManagerContext();
+
+    React.useEffect(() => {
+        return reaction(
```

Comment:
Checking `general_store.props.setP2POrderProps` inside useEffect without reaction is not sufficient, since `useEffect` is unable to detect `setP2POrderProps` action initialization and would return undefined since its a proxy, thus placing it under a reaction ensures it only invokes it when its available